### PR TITLE
Migrate to TanStack Table

### PR DESCRIPTION
**Pull request by neotrow**

### CODE REVIEW

File: src/lib/useReactDataTableState/useReactDataTableState.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  const [pagination, setPagination] = useState<PaginationState>({
+    pageIndex: initialPagination?.pageIndex ?? 0,
+    pageSize: initialPagination?.pageSize ?? 10,
+  });
+
+  return {
+    sorting,
+    pagination,
+    columnFilters,
+    enteredColumnFilters,
```
Comment (neotrow) [inline src/lib/useReactDataTableState/useReactDataTableState.ts:25]: @neotob 

Is the magnifying glass button still being drawn? If yes than i think we should search for a better name for this.

File: src/react-table.d.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+export interface EnumValue {
+  value: number | string | undefined;
+  text: string;
+  disabled?: boolean;
+}
+
+declare module "@tanstack/table-core" {
+  interface ColumnMeta<TData extends RowData, TValue> {
+    dropdownFilter?: DropdownColumnFilter;
```
Comment (neotrow) [inline src/react-table.d.ts:23]: @neotob 

I think something like `filterVariant` would be better? 
Possible options could be: `dropDown`, `textInput`, `numberInput` etc.

What do you think?

File: src/lib/useReactDataTableState/useReactDataTableState.ts (hunk 1 above)
Comment (neotob) [reply src/lib/useReactDataTableState/useReactDataTableState.ts:25]: the searchbutton is still drawn yes, i was thinking about "activeFilter" too :thinking:

Comment (neotrow) [reply src/lib/useReactDataTableState/useReactDataTableState.ts:25]: as discussed: `afterSearchFilter` or `searchedFilter`

Comment (neotob) [reply src/lib/useReactDataTableState/useReactDataTableState.ts]: renamed 💪

File: src/react-table.d.ts (hunk 2 above)
Comment (neotob) [reply src/react-table.d.ts:24]: enum values is removed, so we keep the dropdownFilter.options.

File: src/react-table.d.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  label: string;
+  value: string | number;
+  disabled?: boolean;
+}
+
+interface DropdownColumnFilter {
+  options: DropdownColumnFilterOption[];
+}
+
+export interface EnumValue {
```
Comment (neotob) [inline src/react-table.d.ts:16]: @manni497 i cannot find that this is still used, do you know if we use that somewhere? else i will remove it.

Comment (manni497) [reply src/react-table.d.ts:16]: @neotob Yes, it's not used :D

Comment (neotob) [reply src/react-table.d.ts:16]: removed thanks :)

**Review by neotob — APPROVED** [review]

**Review by manni497 — APPROVED** [review]

**Pull request merged**